sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -505,7 +505,7 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     val schema = StructType(Seq(
       StructField("f", StructType(Seq(
         StructField("a", StringType, nullable = true),
-        StructField("b", IntegerType, nullable = false)
+        StructField("b", IntegerType, nullable = true)
```

Comment:
yea, `row nullability mismatch` checks the error that we pass in a null column while this column is declared as not nullable.